restli-server/src/test/java/com/linkedin/restli/internal/server/response/TestBatchFinderResponseBuilder.java

```diff
@@ -341,4 +344,81 @@ private static Foo generateMetaData(Boolean onError)
     Foo foo = new Foo(map);
     return foo;
   }
+
+  @DataProvider(name = "build_uri_test_cases")
+  public Object[][] testBuildCriteriaURIDataProvider() throws RestLiSyntaxException
+  {
+    String[] requestURIs =
+    {
+        "/greetings?bq=searchGreetings&criteria=List((id:1,tone:SINCERE))&message=hello",
+        "/talent/api/talentHiringProjectCandidates?bq=candidates&candidates=List((candidate:urn%3Ali%3Ats_hire_identity%3A88156577))",
+        "/talent/api/talentHiringProjectCandidates?bq=candidates&candidates=List((candidate:urn%3Ali%3Ats_hire_identity%3A88156577))&fields=candidate",
+        "/talent/api/talentHiringProjectCandidates?bq=candidates&candidates=List((candidate:urn%3Ali%3Ats_hire_identity%3A88156577),(candidate:urn%3Baba%3Ats_hire_identity%3A88156588))",
+        "/talent/api/talentHiringProjectCandidates?bq=candidates&candidates=List((candidate:urn%3Ali%3Ats_hire_identity%3A88156577),(candidate:urn%3Baba%3Ats_hire_identity%3A88156588))&fields=candidate",
+    };
+
+    String[] batchCriteriaParameterNames =
+    {
+        "criteria",
+        "candidates",
+        "candidates",
+        "candidates",
+        "candidates",
+    };
+
+    String[] expectedURIs =
+    {
+        "/greetings?bq=searchGreetings&criteria=List((id:1,tone:SINCERE))&message=hello",
+        "/talent/api/talentHiringProjectCandidates?bq=candidates&candidates=List((candidate:urn%3Ali%3Ats_hire_identity%3A88156577))",
+        "/talent/api/talentHiringProjectCandidates?bq=candidates&candidates=List((candidate:urn%3Ali%3Ats_hire_identity%3A88156577))&fields=candidate",
+        "/talent/api/talentHiringProjectCandidates?bq=candidates&candidates=List((candidate:urn%3Ali%3Ats_hire_identity%3A88156577))",
+        "/talent/api/talentHiringProjectCandidates?bq=candidates&candidates=List((candidate:urn;aba%3Ats_hire_identity%3A88156588))",
+        "/talent/api/talentHiringProjectCandidates?bq=candidates&candidates=List((candidate:urn%3Ali%3Ats_hire_identity%3A88156577))&fields=candidate",
+        "/talent/api/talentHiringProjectCandidates?bq=candidates&candidates=List((candidate:urn;aba%3Ats_hire_identity%3A88156588))&fields=candidate",
```

Comment:
Note the original case's %3B became ;
that is minor but I just want to highlight it in case something is wrong